.buildkite/scripts/publish_docker.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# !!! WARNING DO NOT add -x to avoid leaking vault passwords
+set -euo pipefail
+
+sudo apt-get update
+sudo DEBIAN_FRONTEND=noninteractive apt-get install ca-certificates curl gnupg lsb-release -y
+
+BASEDIR=$(realpath $(dirname $0))
+ROOT=$(realpath $BASEDIR/../)
+
+cd $ROOT
+
+# docker snapshot publication
+echo "Building the image"
+make docker-build
+
+# !!! WARNING be cautious about the following lines, to avoid leaking the secrets in the CI logs
+
+set +x   # Do not remove so we don't leak passwords
+VAULT_ADDR=${VAULT_ADDR:-https://secrets.elastic.co}
+
+echo "Fetching Docker credentials for '$VAULT_USER' from Vault..."
+DOCKER_USER=$(vault read -address "${VAULT_ADDR}" -field user_20230609 secret/ci/elastic-connectors/${VAULT_USER})
+DOCKER_PASSWORD=$(vault read -address "${VAULT_ADDR}" -field secret_20230609 secret/ci/elastic-connectors/${VAULT_USER})
```

Comment:
These need to be replaced with the vault paths and field names for the docker user/password secrets that are specific to Crawler.

Reply:
Idk how I missed this one 🫠